### PR TITLE
CAMEL-18779: Replace ActiveMQ with Artemis

### PR DESCRIPTION
**Pull request by Croway**
This PR update the test-infra-artemis module and the following components:

* camel-jms
* camel-sjms
* camel-amqp

camel-sjms2 will be updated later on.

### CODE REVIEW

File: components/camel-sjms/src/test/java/org/apache/camel/component/sjms/producer/QueueProducerQoSTest.java [hunk 1]
```diff
@@ -106,8 +99,9 @@ public void testInOutQueueProducerTTL() throws Exception {
 
         MockEndpoint.assertIsSatisfied(context);
 
-        DestinationViewMBean queue = service.getQueueMBean(TEST_INOUT_DESTINATION_NAME);
-        assertEquals(0, queue.getQueueSize(),
+        //TODO - I've exposed service.getMBeanServer(), do you think it  can be used here?
```
Comment (orpiske) [inline components/camel-sjms/src/test/java/org/apache/camel/component/sjms/producer/QueueProducerQoSTest.java:102]: Maybe a leftover here?

**Review by orpiske — APPROVED** [review]
Nice one. I've left a few comments, but nothing major.

**Comment by orpiske** [comment]
> ### Components tested:
> Total 	Tested 	Failed ❌ 	Passed ✅
> 7 	7 	1 	6

I haven't had these failures locally when I tried. I suspect they may be flaky on GH only. I am attaching the JMS log (the only component that had failures). 

In this case, I think it may be OK to disable them using the following:

```
@DisabledIfSystemProperty(named = "ci.env.name", matches = "github.com", disabledReason = "Flaky on Github CI")
``` 


[components-camel-jms.log](https://github.com/apache/camel/files/10359024/components-camel-jms.log)



[camel-jms-test.log](https://github.com/apache/camel/files/10359021/camel-jms-test.log)

**Review by davsclaus — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

File: components/camel-sjms/src/test/java/org/apache/camel/component/sjms/producer/QueueProducerQoSTest.java (hunk 1 above)
Comment (Croway) [reply components/camel-sjms/src/test/java/org/apache/camel/component/sjms/producer/QueueProducerQoSTest.java]: Exactly, I tried to re-implement that test with JMX but Artemis expose a ManagementService that did the trick.

**Comment by orpiske** [comment]
> ↩️ There are either **too many** changes to be tested in this PR or the code **needs be rebased**: (42 components likely to be affected)

@Croway please, can you rebase and push again, so that we have a clean test with this?

**Comment by orpiske** [comment]
> ### Components tested:
> Total 	Tested 	Failed ❌ 	Passed ✅
> 3 	3 	1 	2

Failed due to a flaky on:

```
[ERROR] org.apache.camel.component.jms.ManagedJmsEndpointTopicTest.testJmsEndpoint
[ERROR]   Run 1: ManagedJmsEndpointTopicTest.testJmsEndpoint:72 mock://result Received message count. Expected: <2> but was: <1>
[ERROR]   Run 2: ManagedJmsEndpointTopicTest.testJmsEndpoint:72 mock://result Received message count. Expected: <2> but was: <1>
[ERROR]   Run 3: ManagedJmsEndpointTopicTest.testJmsEndpoint:72 mock://result Received message count. Expected: <2> but was: <1>
```

**Comment by Croway** [comment]
> > ### Components tested:
> > Total 	Tested 	Failed x 	Passed white_check_mark
> > 3 	3 	1 	2
> 
> Failed due to a flaky on:
> 
> ```
> [ERROR] org.apache.camel.component.jms.ManagedJmsEndpointTopicTest.testJmsEndpoint
> [ERROR]   Run 1: ManagedJmsEndpointTopicTest.testJmsEndpoint:72 mock://result Received message count. Expected: <2> but was: <1>
> [ERROR]   Run 2: ManagedJmsEndpointTopicTest.testJmsEndpoint:72 mock://result Received message count. Expected: <2> but was: <1>
> [ERROR]   Run 3: ManagedJmsEndpointTopicTest.testJmsEndpoint:72 mock://result Received message count. Expected: <2> but was: <1>
> ```

Exactly, I noticed that a 100 ms timeout fix this flakiness, should I add the timeout?

**Comment by davsclaus** [comment]
yeah since the counters are in mbeans then it can be "off". You can use awailability in the test instead of hardcoded timeout

**Comment by davsclaus** [comment]
I wonder if there is a flaky in sjms

components/camel-sjms test completed successfully: 3 verified / 1 failed

**Comment by Croway** [comment]
`org.apache.camel.processor.PooledExchangeTest.testSameExchange` in camel-core failed on the ci, I've executed it locally and it works

**Comment by davsclaus** [comment]
Ah yeah its how the output is printed, so  / 1 failed is accumulative so it was the same eg in core as you noticed.

**Comment by davsclaus** [comment]
This looks good lets get this merged

**Pull request merged**